client-app/src/examples/todo/TodoPanelModel.js

```diff
@@ -0,0 +1,268 @@
+        this.gridModel.clear();
+        this.gridModel.loadData(tasks);
+
+        if (every(tasks, {complete: true})) {
+            XH.showBanner({message: 'You did it! All of it!'});
+        }
+    }
+
+    createGridModel() {
+        return new GridModel({
+            emptyText: 'No tasks to show...',
+            selModel: {mode: 'multiple'},
+            sizingMode: 'large',
+            enableExport: true,
+            hideHeaders: true,
+            rowBorders: true,
+            showHover: true,
+            sortBy: 'dueDate',
+            groupSortFn: (a, b) => {
+                a = dueDateGroupSort[a];
+                b = dueDateGroupSort[b];
+                return a - b;
+            },
+            contextMenu: [
+                this.editAction,
+                this.toggleCompleteAction,
+                this.deleteAction,
+                '-',
+                ...GridModel.defaultContextMenu
+            ],
+            columns: [
+                {
+                    ...actionCol,
+                    actions: [
+                        {
+                            displayFn: ({record}) => {
+                                const {complete} = record.data;
+
+                                return {
+                                    icon: complete ?
+                                        Icon.checkCircle({prefix: 'fal', className: 'xh-intent-success large-actions'}) :
+                                        Icon.circle({prefix: 'fal', className: 'xh-text-color-muted large-actions'}),
+                                    tooltip: complete ? 'Mark Incomplete' : 'Mark Complete'
+                                };
+                            },
+                            actionFn: ({record}) => {
+                                this.toggleCompleteAsync([record.data]);
+                            }
+                        }
+                    ]
+                },
+                {
+                    field: 'complete',
+                    hidden: true
+                },
+                {
+                    field: 'description',
+                    flex: 1,
+                    cellClass: 'xh-pad',
+                    autoHeight: true
+                },
+                {
+                    field: 'dueDate',
+                    ...localDateCol,
+                    width: 140,
+                    rendererIsComplex: true,
+                    renderer: (v, {record}) => this.dueDateRenderer(v, {record})
+                },
+                {
+                    field: 'dueDateGroup',
+                    hidden: true
+                }
+            ]
+        });
+    }
+
+    info(message) {
+        XH.toast({message});
+    }
+
+    dueDateRenderer(v, {record}) {
+        const overdue = v && v < LocalDate.today() && !record.data.complete,
+            dateStr = fmtCompactDate(v?.date, {
+                sameDayFmt: '[Today]',
```

Comment:
meant to say in our review -- if we are going to be explicit about specifying custom `sameDayFmt` and `distantFmt`, we should also just explicitly specify `nearFmt`.